src/core/cgroup.c

```diff
@@ -774,13 +774,61 @@ void cgroup_oomd_xattr_apply(Unit *u, const char *cgroup_path) {
                 unit_remove_xattr_graceful(u, cgroup_path, "user.oomd_omit");
 }
 
+int cgroup_log_xattr_apply(Unit *u, const char *cgroup_path) {
+        ExecContext *c;
+        char *pattern;
+        /* Use _cleanup_free_ instead of _cleanup_strv_free_ because we need to clean the strv only, not
+         * the strings owned by ExecContext. */
+        _cleanup_free_ char **patterns_strv = NULL;
+        _cleanup_free_ char *patterns_joined = NULL;
+        size_t len;
+        int r;
+
+        assert(u);
+
+        c = unit_get_exec_context(u);
+        if (!c)
+                return -EINVAL;
+
+        if (set_isempty(c->log_filter_allowed_patterns) && set_isempty(c->log_filter_denied_patterns)) {
+                unit_remove_xattr_graceful(u, cgroup_path, "user.journald_log_filter_patterns");
+                return 0;
+        }
+
+        patterns_strv = set_get_strv(c->log_filter_allowed_patterns);
+        if (!patterns_strv)
+                return -ENOMEM;
+
+        r = strv_push(&patterns_strv, (char *)"\xff");
+        if (r < 0)
+                return r;
+
+        SET_FOREACH(pattern, c->log_filter_denied_patterns) {
+                r = strv_push(&patterns_strv, pattern);
+                if (r < 0)
+                        return r;
+        }
+
+        r = strv_make_nulstr(patterns_strv, &patterns_joined, &len);
+        if (r < 0)
+                return r;
+
+        unit_set_xattr_graceful(u, cgroup_path, "user.journald_log_filter_patterns", patterns_joined, len);
+
+        return 0;
+}
+
 static void cgroup_xattr_apply(Unit *u) {
         bool b;
+        int r;
 
         assert(u);
 
         /* The 'user.*' xattrs can be set from a user manager. */
         cgroup_oomd_xattr_apply(u, u->cgroup_path);
+        r = cgroup_log_xattr_apply(u, u->cgroup_path);
+        if (r != 0)
+                (void) log_unit_warning_errno(u, r, "Failed to set user.journald_log_filter_patterns cgroup xattr, ignoring: %m");
```

Comment:
- `cgroup_log_xattr_apply()` returns negative errno on failure, hence please use `if (r < 0)`.
- It is not necessary to voidify the logger. Please drop `(void)`.
- I think debug level is enough. Please downgrade the log level.